src/main/java/org/broadinstitute/dsm/model/PDF/DownloadPDF.java

```diff
@@ -0,0 +1,291 @@
+import org.broadinstitute.ddp.handlers.util.MedicalInfo;
+import org.broadinstitute.ddp.util.GoogleBucket;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.dto.user.UserDto;
+import org.broadinstitute.dsm.files.CoverPDFProcessor;
+import org.broadinstitute.dsm.files.PDFProcessor;
+import org.broadinstitute.dsm.model.ddp.DDPParticipant;
+import org.broadinstitute.dsm.model.ddp.PDF;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.RequestParameter;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.util.DDPRequestUtil;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.broadinstitute.dsm.util.SystemUtil;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Response;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+@Getter
+public class DownloadPDF {
+    private static final String COVER = "cover";
+    private static final String TISSUE = "tissue";
+    private static final String REQUEST = "request";
+    private static final String IRB = "irb";
+    private static final String JSON_START_DATE = "startDate";
+    private static final String JSON_END_DATE = "endDate";
+
+    private String ddpParticipantId;
+    private String configName;
+    private String medicalRecordId;
+    private List<PDF> pdfs;
+    List<String> oncHistoryIDs;
+
+    Logger logger = LoggerFactory.getLogger(DownloadPDF.class);
+
+    public DownloadPDF(@NonNull String requestBody) {
+        JSONObject jsonObject = new JSONObject(requestBody);
+
+        this.ddpParticipantId = (String) jsonObject.get(RequestParameter.DDP_PARTICIPANT_ID);
+        this.configName = null;
+        if (jsonObject.has(RequestParameter.CONFIG_NAME)) {
+            configName = (String) jsonObject.get(RequestParameter.CONFIG_NAME);
+        }
+        this.medicalRecordId = null;
+        if (jsonObject.has("medicalRecordId")) {
+            this.medicalRecordId = (String) jsonObject.get("medicalRecordId");
+        }
+        if (jsonObject.has("requestId")) {
+            this.oncHistoryIDs = new ArrayList<>();
+            JSONArray array = (JSONArray) jsonObject.get("requestId");
+            for (int i = 0; i < array.length(); i++) {
+                this.oncHistoryIDs.add((String) array.get(i));
+            }
+        }
+        if (jsonObject.has("pdfs")) {
+            this.pdfs =Arrays.asList(new Gson().fromJson(jsonObject.getString("pdfs"), PDF[].class));
+        }
+    }
+
+
+    public Object getPDFs(@NonNull Response response, UserDto user, String realm, String requestBody) {
```

Comment:
Class needs to be as much independent as possible, passing `Response` object and manipulating from here does not look good, try to make all `Response` manipulations in `Route` classes.